### PR TITLE
Implement brick friends view for DashboardOverlay

### PR DESCRIPTION
**Pull request by EVAST9919**
- [x] Depends on https://github.com/ppy/osu/pull/9611

Screenshot_1

### CODE REVIEW

**Comment by bdach** [comment]
Mutual friends will have a 2px `--hsl-pink-2` border around the brick:

2020-07-18-193915_346x54_scrot

which as far as I can see is missing here.

**Review by bdach — DISMISSED** [review]
in addition to the above

**Comment by EVAST9919** [comment]
> Mutual friends will have a 2px `--hsl-pink-2` border around the brick:
> 
> 2020-07-18-193915_346x54_scrot
> 
> which as far as I can see is missing here.

Currently there's no way to know (osu-side) if user is a friend

File: osu.Game/Overlays/SocialOverlay.cs [hunk 1]
```diff
@@ -1,242 +0,0 @@
...
-using osu.Game.Overlays.Social;
-using osu.Game.Users;
-using System.Threading;
-using osu.Framework.Allocation;
-using osu.Framework.Extensions.Color4Extensions;
-using osu.Framework.Threading;
-
-namespace osu.Game.Overlays
-{
-    public class SocialOverlay : SearchableListOverlay<SocialTab, SocialSortCriteria, SortDirection>
```
[the remark this answers is not in the document]
Comment (EVAST9919) [reply osu.Game/Overlays/SocialOverlay.cs:25]: I noticed that SocialOverlay used friend panels while refactoring, and also we don't use this overlay anymore, so I removed it.

[the remark this answers is not in the document]
Comment (bdach) [reply osu.Game/Overlays/SocialOverlay.cs:25]: should be split out for sure.

**Review by smoogipoo — APPROVED** [review]
It feels a bit empty since it doesn't have the onhover border, but it matches web so let's go with it for now.

**Pull request merged**